.github/workflows/rhub.yaml

```diff
@@ -0,0 +1,119 @@
+# This workflow is based on R-hub's generic GitHub Actions workflow file.
+# Its canonical location is at
+# https://github.com/r-hub/actions/blob/main/workflows/rhub.yaml
+
+name: R-hub 🌐
+
+on:
+  workflow_call:
+    inputs:
+      config:
+        description: 'A comma separated list of R-hub platforms to use.'
+        type: string
+        # These platforms correspond roughly to CRAN package checks.
+        # More information: https://github.com/insightsengineering/idr-tasks/issues/781
+        # All supported R-hub platforms can be viewed by running: rhub::rhub_platforms()
+        # Container-based platforms (as opposed to VM-based Windows and macOS platforms)
+        # can be also viewed here: https://r-hub.github.io/containers/containers.html
+        default: 'ubuntu-clang,ubuntu-gcc12,gcc13,ubuntu-next,ubuntu-release,macos-arm64,macos,windows,gcc14,atlas,valgrind,intel,nosuggests,mkl'
```

Comment:
let's spend some time on this as I feel this is super important.

This is the current set of what is executed:
![image](https://github.com/insightsengineering/r.pkg.template/assets/12943682/5e535697-702c-4331-97d5-d3afc1ff0a67)

IMHO we should target to cover CRAN checks at least plus some extra as we want

Looking at the [latest results for tern](https://cran.r-project.org/web/checks/check_results_tern.html) we have the following:
![image](https://github.com/insightsengineering/r.pkg.template/assets/12943682/4cf428df-c81d-4fed-aaa8-9cf5b14daa82)

There are some gaps. Testing different versions of R in particular. Looking into the `rhub/actions/setup` there is something about R versions but I haven't cracked parsing the inputs completely I have to admit.

Reply:
In [this issue](https://github.com/insightsengineering/idr-tasks/issues/781) (which I mentioned in a comment in the workflow), I made a summary of different R-hub check names and how I think they correspond to CRAN checks.

Also according to [this JSON](https://github.com/r-hub/actions/blob/main/setup/platforms.json), R-hub and CRAN checks don't seem to correspond to each other perfectly, for example [here](https://github.com/r-hub/actions/blob/main/setup/platforms.json#L201-L205) `macos` R-hub check seems to correspond to both `r-release-macos-x86_64` and `r-oldrel-macos-x86_64` CRAN checks.

Reply:
Not exactly. I tried to debug this functionality and please have a look. I'm debugging `match_platforms()` executed [here](https://github.com/r-hub/actions/blob/a031661736e5ea1c2e35fb27ab406f4150b0a0ef/setup/platforms.R#L183).
```
> match_platforms(list(platforms = "r-oldrel-macos"))
$containers
list()

$platforms
$platforms[[1]]
$platforms[[1]]$label
[1] "macos (r-oldrel-macos) (R-oldrel)"

$platforms[[1]]$name
[1] "macos"

$platforms[[1]]$os
[1] "macos-13"

$platforms[[1]]$`r-version`
[1] "oldrel"

$platforms[[1]]$`job-config`
[1] "{\"label\":\"macos (r-oldrel-macos) (R-oldrel)\",\"name\":\"macos\",\"os\":\"macos-13\",\"r-version\":\"oldrel\"}"
```

```
> match_platforms(list(platforms = "r-release-macos"))
$containers
list()

$platforms
$platforms[[1]]
$platforms[[1]]$label
[1] "macos (r-release-macos) (R-release)"

$platforms[[1]]$name
[1] "macos"

$platforms[[1]]$os
[1] "macos-13"

$platforms[[1]]$`r-version`
[1] "release"

$platforms[[1]]$`job-config`
[1] "{\"label\":\"macos (r-release-macos) (R-release)\",\"name\":\"macos\",\"os\":\"macos-13\",\"r-version\":\"release\"}"
```

```
> match_platforms(list(platforms = "r-release-macos-arm64"))
$containers
list()

$platforms
$platforms[[1]]
$platforms[[1]]$label
[1] "macos-arm64 (r-release-macos-arm64) (R-release)"

$platforms[[1]]$name
[1] "macos-arm64"

$platforms[[1]]$os
[1] "macos-latest"

$platforms[[1]]$`r-version`
[1] "release"

$platforms[[1]]$`job-config`
[1] "{\"label\":\"macos-arm64 (r-release-macos-arm64) (R-release)\",\"name\":\"macos-arm64\",\"os\":\"macos-latest\",\"r-version\":\"release\"}"
```

This produces a different return value. In particular - have a look at `job-config` field. 

It looks like input string is deparsed into a few parts and for one of them (OS): multiple values could be indeed mapped into a single entry but the remaining parts (R version) is also carried forward to the returned value.

Reply:
It looks to me that the list of available choices is as follows:
```
'r-devel-linux-x86_64-debian-clang,r-devel-linux-x86_64-debian-gcc,r-patched-linux-x86_64,r-release-linux-x86_64,ATLAS,C23,clang-ASAN,clang16,clang17,clang18,clang19,gcc13,gcc14,noLD,valgrind,MKL,Intel,donttest,noSuggests,rchk,r-devel-windows-x86_64,r-release-windows-x86_64,r-oldrel-windows-x86_64,r-devel-windows,r-release-windows,r-oldrel-windows,r-release-macos-x86_64,r-oldrel-macos-x86_64,r-release-macos-arm64,r-oldrel-macos-arm64,linux'
```

<details>
  <summary>This is the result of deparsing</summary>
code:

```
x <- 'r-devel-linux-x86_64-debian-clang,r-devel-linux-x86_64-debian-gcc,r-patched-linux-x86_64,r-release-linux-x86_64,ATLAS,C23,clang-ASAN,clang16,clang17,clang18,clang19,gcc13,gcc14,noLD,valgrind,MKL,Intel,donttest,noSuggests,rchk,r-devel-windows-x86_64,r-release-windows-x86_64,r-oldrel-windows-x86_64,r-devel-windows,r-release-windows,r-oldrel-windows,r-release-macos-x86_64,r-oldrel-macos-x86_64,r-release-macos-arm64,r-oldrel-macos-arm64,linux'
sapply(X = strsplit(x, ",")[[1]], \(x) match_platforms(list(platforms = x)), USE.NAMES = TRUE, simplify = FALSE)
```

output:

```
$`r-devel-linux-x86_64-debian-clang`
$`r-devel-linux-x86_64-debian-clang`$containers
$`r-devel-linux-x86_64-debian-clang`$containers[[1]]
$`r-devel-linux-x86_64-debian-clang`$containers[[1]]$label
[1] "ubuntu-clang (r-devel-linux-x86_64-debian-clang)"

$`r-devel-linux-x86_64-debian-clang`$containers[[1]]$name
[1] "ubuntu-clang"

$`r-devel-linux-x86_64-debian-clang`$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/ubuntu-clang:latest"

$`r-devel-linux-x86_64-debian-clang`$containers[[1]]$`job-config`
[1] "{\"label\":\"ubuntu-clang (r-devel-linux-x86_64-debian-clang)\",\"name\":\"ubuntu-clang\",\"container\":\"ghcr.io/r-hub/containers/ubuntu-clang:latest\"}"



$`r-devel-linux-x86_64-debian-clang`$platforms
list()


$`r-devel-linux-x86_64-debian-gcc`
$`r-devel-linux-x86_64-debian-gcc`$containers
$`r-devel-linux-x86_64-debian-gcc`$containers[[1]]
$`r-devel-linux-x86_64-debian-gcc`$containers[[1]]$label
[1] "ubuntu-gcc12 (r-devel-linux-x86_64-debian-gcc)"

$`r-devel-linux-x86_64-debian-gcc`$containers[[1]]$name
[1] "ubuntu-gcc12"

$`r-devel-linux-x86_64-debian-gcc`$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/ubuntu-gcc12:latest"

$`r-devel-linux-x86_64-debian-gcc`$containers[[1]]$`job-config`
[1] "{\"label\":\"ubuntu-gcc12 (r-devel-linux-x86_64-debian-gcc)\",\"name\":\"ubuntu-gcc12\",\"container\":\"ghcr.io/r-hub/containers/ubuntu-gcc12:latest\"}"



$`r-devel-linux-x86_64-debian-gcc`$platforms
list()


$`r-patched-linux-x86_64`
$`r-patched-linux-x86_64`$containers
$`r-patched-linux-x86_64`$containers[[1]]
$`r-patched-linux-x86_64`$containers[[1]]$label
[1] "ubuntu-next (r-patched-linux-x86_64)"

$`r-patched-linux-x86_64`$containers[[1]]$name
[1] "ubuntu-next"

$`r-patched-linux-x86_64`$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/ubuntu-next:latest"

$`r-patched-linux-x86_64`$containers[[1]]$`job-config`
[1] "{\"label\":\"ubuntu-next (r-patched-linux-x86_64)\",\"name\":\"ubuntu-next\",\"container\":\"ghcr.io/r-hub/containers/ubuntu-next:latest\"}"



$`r-patched-linux-x86_64`$platforms
list()


$`r-release-linux-x86_64`
$`r-release-linux-x86_64`$containers
$`r-release-linux-x86_64`$containers[[1]]
$`r-release-linux-x86_64`$containers[[1]]$label
[1] "ubuntu-release (r-release-linux-x86_64)"

$`r-release-linux-x86_64`$containers[[1]]$name
[1] "ubuntu-release"

$`r-release-linux-x86_64`$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/ubuntu-release:latest"

$`r-release-linux-x86_64`$containers[[1]]$`job-config`
[1] "{\"label\":\"ubuntu-release (r-release-linux-x86_64)\",\"name\":\"ubuntu-release\",\"container\":\"ghcr.io/r-hub/containers/ubuntu-release:latest\"}"



$`r-release-linux-x86_64`$platforms
list()


$ATLAS
$ATLAS$containers
$ATLAS$containers[[1]]
$ATLAS$containers[[1]]$label
[1] "atlas (ATLAS)"

$ATLAS$containers[[1]]$name
[1] "atlas"

$ATLAS$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/atlas:latest"

$ATLAS$containers[[1]]$`job-config`
[1] "{\"label\":\"atlas (ATLAS)\",\"name\":\"atlas\",\"container\":\"ghcr.io/r-hub/containers/atlas:latest\"}"



$ATLAS$platforms
list()


$C23
$C23$containers
$C23$containers[[1]]
$C23$containers[[1]]$label
[1] "c23 (C23)"

$C23$containers[[1]]$name
[1] "c23"

$C23$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/c23:latest"

$C23$containers[[1]]$`job-config`
[1] "{\"label\":\"c23 (C23)\",\"name\":\"c23\",\"container\":\"ghcr.io/r-hub/containers/c23:latest\"}"



$C23$platforms
list()


$`clang-ASAN`
$`clang-ASAN`$containers
$`clang-ASAN`$containers[[1]]
$`clang-ASAN`$containers[[1]]$label
[1] "clang-asan (clang-ASAN)"

$`clang-ASAN`$containers[[1]]$name
[1] "clang-asan"

$`clang-ASAN`$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/clang-asan:latest"

$`clang-ASAN`$containers[[1]]$`job-config`
[1] "{\"label\":\"clang-asan (clang-ASAN)\",\"name\":\"clang-asan\",\"container\":\"ghcr.io/r-hub/containers/clang-asan:latest\"}"



$`clang-ASAN`$platforms
list()


$clang16
$clang16$containers
$clang16$containers[[1]]
$clang16$containers[[1]]$label
[1] "clang16"

$clang16$containers[[1]]$name
[1] "clang16"

$clang16$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/clang16:latest"

$clang16$containers[[1]]$`job-config`
[1] "{\"label\":\"clang16\",\"name\":\"clang16\",\"container\":\"ghcr.io/r-hub/containers/clang16:latest\"}"



$clang16$platforms
list()


$clang17
$clang17$containers
$clang17$containers[[1]]
$clang17$containers[[1]]$label
[1] "clang17"

$clang17$containers[[1]]$name
[1] "clang17"

$clang17$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/clang17:latest"

$clang17$containers[[1]]$`job-config`
[1] "{\"label\":\"clang17\",\"name\":\"clang17\",\"container\":\"ghcr.io/r-hub/containers/clang17:latest\"}"



$clang17$platforms
list()


$clang18
$clang18$containers
$clang18$containers[[1]]
$clang18$containers[[1]]$label
[1] "clang18"

$clang18$containers[[1]]$name
[1] "clang18"

$clang18$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/clang18:latest"

$clang18$containers[[1]]$`job-config`
[1] "{\"label\":\"clang18\",\"name\":\"clang18\",\"container\":\"ghcr.io/r-hub/containers/clang18:latest\"}"



$clang18$platforms
list()


$clang19
$clang19$containers
$clang19$containers[[1]]
$clang19$containers[[1]]$label
[1] "clang19"

$clang19$containers[[1]]$name
[1] "clang19"

$clang19$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/clang19:latest"

$clang19$containers[[1]]$`job-config`
[1] "{\"label\":\"clang19\",\"name\":\"clang19\",\"container\":\"ghcr.io/r-hub/containers/clang19:latest\"}"



$clang19$platforms
list()


$gcc13
$gcc13$containers
$gcc13$containers[[1]]
$gcc13$containers[[1]]$label
[1] "gcc13"

$gcc13$containers[[1]]$name
[1] "gcc13"

$gcc13$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/gcc13:latest"

$gcc13$containers[[1]]$`job-config`
[1] "{\"label\":\"gcc13\",\"name\":\"gcc13\",\"container\":\"ghcr.io/r-hub/containers/gcc13:latest\"}"



$gcc13$platforms
list()


$gcc14
$gcc14$containers
$gcc14$containers[[1]]
$gcc14$containers[[1]]$label
[1] "gcc14"

$gcc14$containers[[1]]$name
[1] "gcc14"

$gcc14$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/gcc14:latest"

$gcc14$containers[[1]]$`job-config`
[1] "{\"label\":\"gcc14\",\"name\":\"gcc14\",\"container\":\"ghcr.io/r-hub/containers/gcc14:latest\"}"



$gcc14$platforms
list()


$noLD
$noLD$containers
$noLD$containers[[1]]
$noLD$containers[[1]]$label
[1] "nold (noLD)"

$noLD$containers[[1]]$name
[1] "nold"

$noLD$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/nold:latest"

$noLD$containers[[1]]$`job-config`
[1] "{\"label\":\"nold (noLD)\",\"name\":\"nold\",\"container\":\"ghcr.io/r-hub/containers/nold:latest\"}"



$noLD$platforms
list()


$valgrind
$valgrind$containers
$valgrind$containers[[1]]
$valgrind$containers[[1]]$label
[1] "valgrind"

$valgrind$containers[[1]]$name
[1] "valgrind"

$valgrind$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/valgrind:latest"

$valgrind$containers[[1]]$`job-config`
[1] "{\"label\":\"valgrind\",\"name\":\"valgrind\",\"container\":\"ghcr.io/r-hub/containers/valgrind:latest\"}"



$valgrind$platforms
list()


$MKL
$MKL$containers
$MKL$containers[[1]]
$MKL$containers[[1]]$label
[1] "mkl (MKL)"

$MKL$containers[[1]]$name
[1] "mkl"

$MKL$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/mkl:latest"

$MKL$containers[[1]]$`job-config`
[1] "{\"label\":\"mkl (MKL)\",\"name\":\"mkl\",\"container\":\"ghcr.io/r-hub/containers/mkl:latest\"}"



$MKL$platforms
list()


$Intel
$Intel$containers
$Intel$containers[[1]]
$Intel$containers[[1]]$label
[1] "intel (Intel)"

$Intel$containers[[1]]$name
[1] "intel"

$Intel$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/intel:latest"

$Intel$containers[[1]]$`job-config`
[1] "{\"label\":\"intel (Intel)\",\"name\":\"intel\",\"container\":\"ghcr.io/r-hub/containers/intel:latest\"}"



$Intel$platforms
list()


$donttest
$donttest$containers
$donttest$containers[[1]]
$donttest$containers[[1]]$label
[1] "donttest"

$donttest$containers[[1]]$name
[1] "donttest"

$donttest$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/donttest:latest"

$donttest$containers[[1]]$`job-config`
[1] "{\"label\":\"donttest\",\"name\":\"donttest\",\"container\":\"ghcr.io/r-hub/containers/donttest:latest\"}"



$donttest$platforms
list()


$noSuggests
$noSuggests$containers
$noSuggests$containers[[1]]
$noSuggests$containers[[1]]$label
[1] "nosuggests (noSuggests)"

$noSuggests$containers[[1]]$name
[1] "nosuggests"

$noSuggests$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/nosuggests:latest"

$noSuggests$containers[[1]]$`job-config`
[1] "{\"label\":\"nosuggests (noSuggests)\",\"name\":\"nosuggests\",\"container\":\"ghcr.io/r-hub/containers/nosuggests:latest\"}"



$noSuggests$platforms
list()


$rchk
$rchk$containers
$rchk$containers[[1]]
$rchk$containers[[1]]$label
[1] "rchk"

$rchk$containers[[1]]$name
[1] "rchk"

$rchk$containers[[1]]$container
[1] "ghcr.io/r-hub/containers/rchk:latest"

$rchk$containers[[1]]$`job-config`
[1] "{\"label\":\"rchk\",\"name\":\"rchk\",\"container\":\"ghcr.io/r-hub/containers/rchk:latest\"}"



$rchk$platforms
list()


$`r-devel-windows-x86_64`
$`r-devel-windows-x86_64`$containers
list()

$`r-devel-windows-x86_64`$platforms
$`r-devel-windows-x86_64`$platforms[[1]]
$`r-devel-windows-x86_64`$platforms[[1]]$label
[1] "windows (r-devel-windows-x86_64) (R-devel)"

$`r-devel-windows-x86_64`$platforms[[1]]$name
[1] "windows"

$`r-devel-windows-x86_64`$platforms[[1]]$os
[1] "windows-latest"

$`r-devel-windows-x86_64`$platforms[[1]]$`r-version`
[1] "devel"

$`r-devel-windows-x86_64`$platforms[[1]]$`job-config`
[1] "{\"label\":\"windows (r-devel-windows-x86_64) (R-devel)\",\"name\":\"windows\",\"os\":\"windows-latest\",\"r-version\":\"devel\"}"




$`r-release-windows-x86_64`
$`r-release-windows-x86_64`$containers
list()

$`r-release-windows-x86_64`$platforms
$`r-release-windows-x86_64`$platforms[[1]]
$`r-release-windows-x86_64`$platforms[[1]]$label
[1] "windows (r-release-windows-x86_64) (R-release)"

$`r-release-windows-x86_64`$platforms[[1]]$name
[1] "windows"

$`r-release-windows-x86_64`$platforms[[1]]$os
[1] "windows-latest"

$`r-release-windows-x86_64`$platforms[[1]]$`r-version`
[1] "release"

$`r-release-windows-x86_64`$platforms[[1]]$`job-config`
[1] "{\"label\":\"windows (r-release-windows-x86_64) (R-release)\",\"name\":\"windows\",\"os\":\"windows-latest\",\"r-version\":\"release\"}"




$`r-oldrel-windows-x86_64`
$`r-oldrel-windows-x86_64`$containers
list()

$`r-oldrel-windows-x86_64`$platforms
$`r-oldrel-windows-x86_64`$platforms[[1]]
$`r-oldrel-windows-x86_64`$platforms[[1]]$label
[1] "windows (r-oldrel-windows-x86_64) (R-oldrel)"

$`r-oldrel-windows-x86_64`$platforms[[1]]$name
[1] "windows"

$`r-oldrel-windows-x86_64`$platforms[[1]]$os
[1] "windows-latest"

$`r-oldrel-windows-x86_64`$platforms[[1]]$`r-version`
[1] "oldrel"

$`r-oldrel-windows-x86_64`$platforms[[1]]$`job-config`
[1] "{\"label\":\"windows (r-oldrel-windows-x86_64) (R-oldrel)\",\"name\":\"windows\",\"os\":\"windows-latest\",\"r-version\":\"oldrel\"}"




$`r-devel-windows`
$`r-devel-windows`$containers
list()

$`r-devel-windows`$platforms
$`r-devel-windows`$platforms[[1]]
$`r-devel-windows`$platforms[[1]]$label
[1] "windows (r-devel-windows) (R-devel)"

$`r-devel-windows`$platforms[[1]]$name
[1] "windows"

$`r-devel-windows`$platforms[[1]]$os
[1] "windows-latest"

$`r-devel-windows`$platforms[[1]]$`r-version`
[1] "devel"

$`r-devel-windows`$platforms[[1]]$`job-config`
[1] "{\"label\":\"windows (r-devel-windows) (R-devel)\",\"name\":\"windows\",\"os\":\"windows-latest\",\"r-version\":\"devel\"}"




$`r-release-windows`
$`r-release-windows`$containers
list()

$`r-release-windows`$platforms
$`r-release-windows`$platforms[[1]]
$`r-release-windows`$platforms[[1]]$label
[1] "windows (r-release-windows) (R-release)"

$`r-release-windows`$platforms[[1]]$name
[1] "windows"

$`r-release-windows`$platforms[[1]]$os
[1] "windows-latest"

$`r-release-windows`$platforms[[1]]$`r-version`
[1] "release"

$`r-release-windows`$platforms[[1]]$`job-config`
[1] "{\"label\":\"windows (r-release-windows) (R-release)\",\"name\":\"windows\",\"os\":\"windows-latest\",\"r-version\":\"release\"}"




$`r-oldrel-windows`
$`r-oldrel-windows`$containers
list()

$`r-oldrel-windows`$platforms
$`r-oldrel-windows`$platforms[[1]]
$`r-oldrel-windows`$platforms[[1]]$label
[1] "windows (r-oldrel-windows) (R-oldrel)"

$`r-oldrel-windows`$platforms[[1]]$name
[1] "windows"

$`r-oldrel-windows`$platforms[[1]]$os
[1] "windows-latest"

$`r-oldrel-windows`$platforms[[1]]$`r-version`
[1] "oldrel"

$`r-oldrel-windows`$platforms[[1]]$`job-config`
[1] "{\"label\":\"windows (r-oldrel-windows) (R-oldrel)\",\"name\":\"windows\",\"os\":\"windows-latest\",\"r-version\":\"oldrel\"}"




$`r-release-macos-x86_64`
$`r-release-macos-x86_64`$containers
list()

$`r-release-macos-x86_64`$platforms
$`r-release-macos-x86_64`$platforms[[1]]
$`r-release-macos-x86_64`$platforms[[1]]$label
[1] "macos (r-release-macos-x86_64) (R-release)"

$`r-release-macos-x86_64`$platforms[[1]]$name
[1] "macos"

$`r-release-macos-x86_64`$platforms[[1]]$os
[1] "macos-13"

$`r-release-macos-x86_64`$platforms[[1]]$`r-version`
[1] "release"

$`r-release-macos-x86_64`$platforms[[1]]$`job-config`
[1] "{\"label\":\"macos (r-release-macos-x86_64) (R-release)\",\"name\":\"macos\",\"os\":\"macos-13\",\"r-version\":\"release\"}"




$`r-oldrel-macos-x86_64`
$`r-oldrel-macos-x86_64`$containers
list()

$`r-oldrel-macos-x86_64`$platforms
$`r-oldrel-macos-x86_64`$platforms[[1]]
$`r-oldrel-macos-x86_64`$platforms[[1]]$label
[1] "macos (r-oldrel-macos-x86_64) (R-oldrel)"

$`r-oldrel-macos-x86_64`$platforms[[1]]$name
[1] "macos"

$`r-oldrel-macos-x86_64`$platforms[[1]]$os
[1] "macos-13"

$`r-oldrel-macos-x86_64`$platforms[[1]]$`r-version`
[1] "oldrel"

$`r-oldrel-macos-x86_64`$platforms[[1]]$`job-config`
[1] "{\"label\":\"macos (r-oldrel-macos-x86_64) (R-oldrel)\",\"name\":\"macos\",\"os\":\"macos-13\",\"r-version\":\"oldrel\"}"




$`r-release-macos-arm64`
$`r-release-macos-arm64`$containers
list()

$`r-release-macos-arm64`$platforms
$`r-release-macos-arm64`$platforms[[1]]
$`r-release-macos-arm64`$platforms[[1]]$label
[1] "macos-arm64 (r-release-macos-arm64) (R-release)"

$`r-release-macos-arm64`$platforms[[1]]$name
[1] "macos-arm64"

$`r-release-macos-arm64`$platforms[[1]]$os
[1] "macos-latest"

$`r-release-macos-arm64`$platforms[[1]]$`r-version`
[1] "release"

$`r-release-macos-arm64`$platforms[[1]]$`job-config`
[1] "{\"label\":\"macos-arm64 (r-release-macos-arm64) (R-release)\",\"name\":\"macos-arm64\",\"os\":\"macos-latest\",\"r-version\":\"release\"}"




$`r-oldrel-macos-arm64`
$`r-oldrel-macos-arm64`$containers
list()

$`r-oldrel-macos-arm64`$platforms
$`r-oldrel-macos-arm64`$platforms[[1]]
$`r-oldrel-macos-arm64`$platforms[[1]]$label
[1] "macos-arm64 (r-oldrel-macos-arm64) (R-oldrel)"

$`r-oldrel-macos-arm64`$platforms[[1]]$name
[1] "macos-arm64"

$`r-oldrel-macos-arm64`$platforms[[1]]$os
[1] "macos-latest"

$`r-oldrel-macos-arm64`$platforms[[1]]$`r-version`
[1] "oldrel"

$`r-oldrel-macos-arm64`$platforms[[1]]$`job-config`
[1] "{\"label\":\"macos-arm64 (r-oldrel-macos-arm64) (R-oldrel)\",\"name\":\"macos-arm64\",\"os\":\"macos-latest\",\"r-version\":\"oldrel\"}"




$linux
$linux$containers
list()

$linux$platforms
$linux$platforms[[1]]
$linux$platforms[[1]]$label
[1] "linux (R-devel)"

$linux$platforms[[1]]$name
[1] "linux"

$linux$platforms[[1]]$os
[1] "ubuntu-latest"

$linux$platforms[[1]]$`r-version`
[1] "devel"

$linux$platforms[[1]]$`job-config`
[1] "{\"label\":\"linux (R-devel)\",\"name\":\"linux\",\"os\":\"ubuntu-latest\",\"r-version\":\"devel\"}"
```
</details>

Reply:
Looks like this is working! https://github.com/insightsengineering/teal/actions/runs/9797793526

Now we need to double check against duplicates

Reply:
Okay, then maybe this list could look like this?
```yaml
config: >-
  r-devel-linux-x86_64-debian-clang,
  r-devel-linux-x86_64-debian-gcc,
  gcc14,
  r-devel-windows-x86_64,
  r-patched-linux-x86_64,
  r-release-linux-x86_64,
  r-release-macos-arm64,
  r-release-macos-x86_64,
  r-release-windows-x86_64,
  r-oldrel-macos-arm64,
  r-oldrel-macos-x86_64,
  r-oldrel-windows-x86_64
```
Almost all CRAN checks are here (`gcc14` is Fedora with gcc which corresponds to `r-devel-linux-x86_64-fedora-gcc`). The only one I still couldn't find is Fedora with clang (`r-devel-linux-x86_64-fedora-clang`).

Reply:
The above list would nicely cover CRAN (except that one you noticed). Let's also add something at the top. I was thinking about `linux` (Ubuntu), `noSuggests`, `donttest` and `gcc13`.

We can add more but I honestly don't know about them or don't see much value. Please also think on your side which would be good to add.

Reply:
```suggestion
        default: 'r-devel-linux-x86_64-debian-clang, r-devel-linux-x86_64-debian-gcc, gcc14, r-devel-windows-x86_64, r-patched-linux-x86_64, r-release-linux-x86_64, r-release-macos-arm64, r-release-macos-x86_64, r-release-windows-x86_64, r-oldrel-macos-arm64, r-oldrel-macos-x86_64, r-oldrel-windows-x86_64, linux, gc13, noSuggests, donttest'
```